### PR TITLE
Support activation hooks by grammar scope

### LINKED ISSUE #14148: Support for grammar scopeName based activation hooks
At the moment Atom supports `activationHooks` like `${packageName}:grammar-used`, but these only work on package level and not on the level of the actual grammar. This has two downsides:
1. If a package defines multiple grammars, the `activationHooks` will be triggered for any grammar. An example for this is the `language-java` package.
2. if multiple packages define the same grammar, a package that should be activated if this grammar is used has to add `activationHooks` for every package and update the `activationHooks` if a new package implements that grammar. An example for this is the `language-javascript` and `language-babel` package.

I think it wound be best to have another activation hook that works on the `scopeName` of a grammar. Something like: `${scopeName}:grammar-scope-used`.

### PR DESCRIPTION
**Pull request by Aerijo**
### Description of the Change

Adds activation hook for grammar scope name.

### Alternate Designs

None

### Why Should This Be In Core?

Is core mechanic

### Benefits

Makes activation package agnostic, much like the services API.

### Possible Drawbacks

None

### Verification Process

Tested manually. ~Will add tests.~

Test added.

### Applicable Issues

Fixes #14148

### CODE REVIEW

File: src/workspace.js [hunk 1]
```diff
@@ -1268,6 +1268,7 @@ module.exports = class Workspace extends Model {
 
   handleGrammarUsed (grammar) {
     if (grammar == null) { return }
+    this.packageManager.triggerActivationHook(`${grammar.scopeName}:root-scope-used`)
     return this.packageManager.triggerActivationHook(`${grammar.packageName}:grammar-used`)
```
Comment (thomasjo) [inline src/workspace.js]: Outside the scope of this PR, but remarking on it anyway; should we :fire the `return` here? There is no value returned by `triggerActivationHook`, so returning here doesn't make any sense and is simply confusing.

Comment (Aerijo) [reply src/workspace.js]: I'd remove it. Should be fine to do it in this PR right?

Comment (thomasjo) [reply src/workspace.js]: Yeah, definitely fine in my opinion. Just didn't want to expand the scope of your PR :sweat_smile:

**Comment by Aerijo** [comment]
The test is just a copy of the one above, but the hook changed to `source.js:root-scope-used` etc.

I think it passes, but running it on my computer seemed to always cause a timeout on the promises (for this, the one above, and several others that changed on each run).

**Review by thomasjo — APPROVED** [review]
LGTM :ship:

**Review by daviwil — APPROVED** [review]
Thanks a lot @Aerijo, looks great!

**Pull request merged**